sql/src/main/java/org/apache/druid/sql/calcite/view/DruidViewMacro.java

```diff
@@ -58,7 +58,8 @@ public TranslatableTable apply(final List<Object> arguments)
   {
     final RelDataType rowType;
     try (final DruidPlanner planner = plannerFactory.createPlanner(viewSql, new QueryContext())) {
-      rowType = planner.plan().rowType();
+      planner.validate();
+      rowType = planner.prepare().getRowType();
```

Comment:
any reason to switch to prepare row type or just to do a bit less work?

Reply:
I *think* this is due to the state machine added to the planner. It enforces that authorization is done before planning. However, in this context, we have no auth context. So, by using prepare instead, we get the information without the need to authorize.

And, as you point out, the job of prepare is to provide this info: planning (to get a physical execution plan) is overkill.